src/control/zoom/ZoomControl.cpp

```diff
@@ -351,6 +351,16 @@ void ZoomControl::pageSizeChanged(size_t page) {
 }
 
 void ZoomControl::pageSelected(size_t page) {
-    updateZoomPresentationValue(page);
-    updateZoomFitValue(page);
+    // Todo (fabian): page selected should do nothing here, since Zoom Controls, which page is selected.
+    //               This results in a logical loop
+    if (current_page != page) {
+        this->last_page = this->current_page;
+        this->current_page = page;
+    }
+
+    if (view->isPageVisible(this->last_page, nullptr)) {
+        return;
+    }
+    updateZoomPresentationValue(this->current_page);
```

Comment:
I'm guessing that this should be moved before the visibility check to address the presentation mode zoom problem.

Reply:
done.